src/Kunstmaan/FormBundle/Controller/FormSubmissionsController.php

```diff
@@ -89,11 +89,16 @@ public function listAction(Request $request, $nodeTranslationId)
      */
     public function editAction($nodeTranslationId, $submissionId)
     {
-        $em              = $this->getDoctrine()->getManager();
-        $nodeTranslation = $em->getRepository('KunstmaanNodeBundle:NodeTranslation')->find($nodeTranslationId);
-        $formSubmission  = $em->getRepository('KunstmaanFormBundle:FormSubmission')->find($submissionId);
+        $em                   = $this->getDoctrine()->getManager();
+        $nodeTranslation      = $em->getRepository('KunstmaanNodeBundle:NodeTranslation')->find($nodeTranslationId);
+        $formSubmission       = $em->getRepository('KunstmaanFormBundle:FormSubmission')->find($submissionId);
+        $formSubmissionFields = $em->getRepository('KunstmaanFormBundle:FormSubmissionField')->getFieldsOrderedBySequence($formSubmission);
```

Comment:
Why would you fetch them separately? The fields are a property of FormSubmission, and can be sorted there according to their sequence by adding a simple Doctrine annotation. cfr. http://doctrine-orm.readthedocs.org/en/latest/tutorials/ordered-associations.html